### ISSUE TITLE
NPE when unmarshall exception content when handling a response from Amazon SageMaker endpoint

### ISSUE THREAD

**Issue by azagniotov** [aws/aws-sdk-java#2369]
## Describe the bug
NPE when unmarshall exception content when handling the response from Amazon SageMaker endpoint

## Expected Behavior
No NPE is thrown when unmarshalling exception

## Current Behavior
```
2020-06-24 16:36:57.511 [XXXXXXXX] [INFO ] [c.a.http.JsonErrorResponseHandler       ] - Unable to unmarshall exception content
java.lang.NullPointerException: null
	at com.fasterxml.jackson.databind.ObjectMapper.treeToValue(ObjectMapper.java:2745)
	at com.amazonaws.transform.JsonErrorUnmarshaller.unmarshall(JsonErrorUnmarshaller.java:52)
	at com.amazonaws.http.JsonErrorResponseHandler.doLegacyUnmarshall(JsonErrorResponseHandler.java:185)
	at com.amazonaws.http.JsonErrorResponseHandler.unmarshallException(JsonErrorResponseHandler.java:147)
	at com.amazonaws.http.JsonErrorResponseHandler.createException(JsonErrorResponseHandler.java:131)
	at com.amazonaws.http.JsonErrorResponseHandler.handle(JsonErrorResponseHandler.java:94)
	at com.amazonaws.http.JsonErrorResponseHandler.handle(JsonErrorResponseHandler.java:40)
	at com.amazonaws.http.AwsErrorResponseHandler.handleAse(AwsErrorResponseHandler.java:53)
	at com.amazonaws.http.AwsErrorResponseHandler.handle(AwsErrorResponseHandler.java:41)
	at com.amazonaws.http.AwsErrorResponseHandler.handle(AwsErrorResponseHandler.java:26)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1694)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1367)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1113)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:770)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:744)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.doInvoke(AmazonSageMakerRuntimeClient.java:289)
	at com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.invoke(AmazonSageMakerRuntimeClient.java:256)
	at com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.invoke(AmazonSageMakerRuntimeClient.java:245)
	at com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.executeInvokeEndpoint(AmazonSageMakerRuntimeClient.java:211)
	at com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.invokeEndpoint(AmazonSageMakerRuntimeClient.java:182)
```

## Steps to Reproduce
N/A This has happened to me only once

## Your Environment

* AWS Java SDK version used:
`aws-java-sdk v1.11.670`

* JDK version used:
```
openjdk version "1.8.0_231"
OpenJDK Runtime Environment (Zulu 8.41.0.4-SA-linux64) (build 1.8.0_231-b08)
OpenJDK 64-Bit Server VM (Zulu 8.41.0.4-SA-linux64) (build 25.231-b08, mixed mode)
```
* Operating System and version:
`CentOS Linux release 7.8.2003 (Core)`

**Comment by debora-ito** [comment]
@azagniotov can you give us more details about your use case?
This has happened once, what makes it different from the other times?
Can you reproduce it? If so, can you enable [wire logs](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-logging.html#sdk-net-logging-verbose)? Make sure there's no sensitive data if you provide us the logs.

**Comment by azagniotov** [comment]
Hi @debora-ito ,

The application that makes requests to Amazon SageMaker has been running in production for the past two weeks and this is the first time that the NPE happened. I cannot reproduce it and I cannot enable wirelogs, unfortunately. I did a little bit extra digging though:

This particular SageMaker request that observed the NPE, has timed out as the `sdkClientExecutionTimeout` was reached. The timeout was configured on the  `AmazonWebServiceRequest` through the extending class `InvokeEndpointRequest` (i.e.: `public <T extends AmazonWebServiceRequest> T withSdkClientExecutionTimeout(int sdkClientExecutionTimeout)`).

This is not the first time that requests timeout due to hitting the aforementioned configured timeout, but this is the first time that the timeout exception (as follows) was also preceded by the NPE. The following is the stacktrace that got logged `168` milliseconds `after` the stacktrace I posted earlier:

```
2020-06-24 16:36:57.679 [XXXXXXXX] [ERROR] [XXXXXXXXXX] - 
Client execution did not complete before the specified timeout configuration.
com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleInterruptedException(AmazonHttpClient.java:831)
com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:728)
com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.doInvoke(AmazonSageMakerRuntimeClient.java:289)
com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.invoke(AmazonSageMakerRuntimeClient.java:256)
com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.invoke(AmazonSageMakerRuntimeClient.java:245)
com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.executeInvokeEndpoint(AmazonSageMakerRuntimeClient.java:211)
com.amazonaws.services.sagemakerruntime.AmazonSageMakerRuntimeClient.invokeEndpoint(AmazonSageMakerRuntimeClient.java:182)
scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:748)
```

I also combed through the Amazon SageMaker endpoint logs, but I could not find any exceptions/error/weird looking stuff for the time preceding the timestamps in the aforementioned stacktraces.

That's all I have.

**Comment by debora-ito** [comment]
@azagniotov apologies for loosing track of this issue after a long time. Without a repro case is hard for us to identify the root cause and fix it. 

Please reply here if you still want me to investigate it, otherwise this issue will auto close soon.

**Comment by azagniotov** [comment]
Well.... It is better after a long time to than never! I will close the issue as this is no longer relevant. Regards
